extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.Optimizely.js

```diff
@@ -0,0 +1,19 @@
+/*!
+ * Utility functions for running Optimizely A/B tests with VisualEditor
+ */
+
+( function () {
+	/**
+	 * @returns {number} The A/B test variant number for the Toolbar A/B test
+	 */
+	ve.init.wikia.getToolbarABTestVariantNumber = function () {
+		var optimizely = window.optimizely,
+			optimizelyId = window.wgDevelEnvironment ? '4721410313' : '4701112678';
+
+		if (optimizely && optimizely.variationMap && optimizely.variationMap.hasOwnProperty(optimizelyId)) {
+			return optimizely.variationMap[optimizelyId] || 0;
```

Comment:
Is this a proper documented way of retrieving variation from Optimizely? I can't see that in their docs - the most similar one I see is: ```optimizely.data.state.variationMap```
Btw. in Optimizely you can configure different variations to execute different JS when activated.
So for instance variation 1 could set some global variable to different value than variation 2. That's what we used to do in the past - worked great - but we were not experimenting too much.